### PR TITLE
1) Fix DECOFLAG order. 2) The doll is Staryu, not Starmie

### PR DESCRIPTION
**Pull request by Tiddlywinkss**
DECOFLAG is used as an index into DecorationIDs, and that list puts big dolls after regular dolls (which does not match DecorationAttributes, which DECO is the index into). Refer to [MysteryGiftGetDecoration](https://github.com/pret/pokegold/blob/master/engine/link/mystery_gift_2.asm#L134), [CheckAndSetMysteryGiftDecorationAlreadyReceived](https://github.com/pret/pokegold/blob/master/engine/link/mystery_gift.asm#L1147), [CopyMysteryGiftReceivedDecorationsToPC](https://github.com/pret/pokegold/blob/master/engine/link/mystery_gift.asm#L1170), [SetSpecificDecorationFlag](https://github.com/pret/pokegold/blob/master/engine/overworld/decorations.asm#L928), [GetDecorationID](https://github.com/pret/pokegold/blob/master/engine/overworld/decorations.asm#L935).

SPRITE_STARMIE is for Staryu AND Starmie, STARMIE_DOLL is just misleading.

### CODE REVIEW

**Comment by mid-kid** [comment]
does this apply to crystal as well?

**Comment by Tiddlywinkss** [comment]
Currently, MysteryGiftDecos [GS](https://github.com/pret/pokegold/blob/master/data/decorations/mystery_gift_decos.asm)/[C](https://github.com/pret/pokecrystal/blob/master/data/decorations/mystery_gift_decos.asm), DecorationIDs [GS](https://github.com/pret/pokegold/blob/master/data/decorations/decorations.asm)/[C](https://github.com/pret/pokecrystal/blob/master/data/decorations/decorations.asm), and the decoration constants [GS](https://github.com/pret/pokegold/blob/master/constants/deco_constants.asm)/[C](https://github.com/pret/pokecrystal/blob/master/constants/deco_constants.asm) are all the same. The functions referenced above are also the same in pokecrystal. The simplest way to see the problem really is choose an index in MysteryGiftDecos, say MysteryGiftDecos[0]: DECOFLAG_PIKACHU_DOLL (22); then you look at DecorationIDs[22]: DECO_JIGGLYPUFF_DOLL (33 fwiw). That switching in the labels is a clear sign one of them is misordered, and that's the same in GS/C. (And FWIW, something like Pink Bed indexes fine through both because it comes before the misordering starts.)

TLDR: yes

**Comment by vulcandth** [comment]
Already merged in 1a5c2b0414dde3a256e675adc914442ecfd192bc.
